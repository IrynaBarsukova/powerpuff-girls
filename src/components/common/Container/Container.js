import React from 'react';
import PropTypes from 'prop-types';

import styles from './Container.scss';

const Container = ({ children }) => {
    return (
        <div className={styles.container}>
            {children}
        </div>
    );
};

Container.propTypes = {
    /**
     * Container's inner content
     */
    children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};

export default Container;
