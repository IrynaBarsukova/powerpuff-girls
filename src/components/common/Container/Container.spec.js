import React from 'react';
import { render } from '@testing-library/react';

import Container from './Container';

describe('<Container />', () => {
    it('should render Container', () => {
        const { container } = render(<Container />);

        expect(container).toMatchInlineSnapshot(`
            <div>
              <div
                class="container"
              />
            </div>
        `);
    });

    it('should render inner content', () => {
        const text = <p>Test text</p>;
        const { getByText } = render(<Container>{text}</Container>);

        expect(getByText('Test text')).toBeInTheDocument();
    });
});
