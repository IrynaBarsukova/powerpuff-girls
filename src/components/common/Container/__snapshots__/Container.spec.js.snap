// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render Container 1`] = `
<div>
  <div
    class="container"
  />
</div>
`;
